cli/api/ioutil2/ioutil.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ioutil2 provides extra functionality along similar lines to io/ioutil.
+package ioutil2
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+// Write file to temp and atomically move when everything else succeeds.
+func WriteFileAtomic(filename string, data []byte, perm os.FileMode) error {
+	dir, name := path.Split(filename)
```

Comment:
Dont we need to check if dir and name are valid? i.e. not empty?

Reply:
n.b. this is code from https://github.com/youtube/vitess/blob/master/go/ioutil2/ioutil.go

I didn't want to import that whole project. That will fail below (and won't happen to us, I think)